no-publish/src/main/scala/org/typelevel/sbt/NoPublishPlugin.scala

```diff
@@ -30,7 +30,11 @@ object NoPublishPlugin extends AutoPlugin {
     publishLocal := {},
     publishArtifact := false,
     publish / skip := true,
-    Global / noPublishInternalAggregation += thisProjectRef.value
+    Global / noPublishModulesIgnore ++= crossScalaVersions.value.map { v =>
+      // the binary versions are needed for the modules-ignore in Submit Dependencies
+      // it's best to pick them up here instead of guessing in the CI plugin
+      s"${thisProjectRef.value.project}_${CrossVersion.binaryScalaVersion(v)}"
```

Comment:
Unfortunately this is not sufficiently general. You should do something like this.
https://github.com/typelevel/sbt-typelevel/blob/fac0ec25fd0644c29238c0de9dfffdc8004a308b/sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala#L78-L82

Reply:
btw, to dog-food this we can add some more no-publish modules to sbt-typelevel's own build.

Maybe one enabling the Scala.js plugin, one disable cross versions (i.e. a java module) and another enabling full cross versions (i.e. a compiler plugin)

Reply:
can the `crossVersion` be assumed to be the same across all the cross-versions?

Reply:
Yes, I suppose that has to be the assumption, otherwise everything breaks down 😅